File: app/src/main/java/com/example/yuxibing/mykotlindemo/javacode/MainActivity.java
package com.example.yuxibing.mykotlindemo.javacode;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

import com.example.yuxibing.mykotlindemo.R;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private CustomView pie;
    private int[] colors = {Color.RED,Color.BLUE,Color.GRAY,Color.GREEN,Color.BLACK};
    private Button btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        pie = (CustomView) findViewById(R.id.pie);
        btn = (Button) findViewById(R.id.btn);
        ArrayList<PieEntity> pieEntities = new ArrayList<>();
        for (int i = 0; i < colors.length; i++) {
            pieEntities.add(new PieEntity(i+1,colors[i]));
        }
        pie.setData(pieEntities);



        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(),ViewGroupActivity.class));
            }
        });
    }

}

File: app/src/main/java/com/example/yuxibing/mykotlindemo/kotlincode/MainActivity.kt
package com.example.yuxibing.mykotlindemo.kotlincode

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.yuxibing.mykotlindemo.R

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val array = ArrayList<Int>()
        for ((index, value) in array.withIndex()) {
            println("the element at $index is $value")
        }
        joinOptions(listOf("ues","no","maybe"))

    }

    fun maxof(a: Int, b: Int) : Int{
        if (a > b) {
            return a
        } else {
            return b
        }
    }
    fun minof(a:Int,b: Int) = if (a>b) a else b
    fun joinOptions(options : Collection<String>) = options.joinToString(prefix = "[",postfix = "]")

}
